src/interface/obsidian/src/chat_modal.ts

```diff
@@ -228,15 +233,33 @@ export class KhojChatModal extends Modal {
         return `${time_string}, ${date_string}`;
     }
 
-    async getChatHistory(chatBodyEl: Element): Promise<void> {
+    async getChatHistory(chatBodyEl: Element): Promise<boolean> {
         // Get chat history from Khoj backend
         let chatUrl = `${this.setting.khojUrl}/api/chat/history?client=obsidian`;
         let headers = { "Authorization": `Bearer ${this.setting.khojApiKey}` };
-        let response = await request({ url: chatUrl, headers: headers });
-        let chatLogs = JSON.parse(response).response;
-        chatLogs.forEach((chatLog: any) => {
-            this.renderMessageWithReferences(chatBodyEl, chatLog.message, chatLog.by, chatLog.context, new Date(chatLog.created), chatLog.intent?.type);
-        });
+
+        try {
+            let response = await fetch(chatUrl, { method: "GET", headers: headers });
+            let responseJson: any = await response.json();
+
+            if (responseJson.detail) {
+                // If the server returns error details in response, render a setup hint.
+                let setupMsg = "Hi 👋🏾, to start chatting add available chat models options via <a class='inline-chat-link' href='/server/admin'>the Django Admin panel</a> on the Server";
```

Comment:
Shouldn't this be changed to use markdown format?

Reply:
Done